nntrainer/layers/cl_layers/fc_layer_cl.cpp

```diff
@@ -0,0 +1,616 @@
+/**
+ * Copyright (C) 2020 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * @file	fc_layer_cl.cpp
+ * @date	7 May 2020
+ * @brief	This is Fully Connected Layer Class for Neural Network with OpenCl
+ * implementation
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Debadri Samaddar <s.debadri@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+
+#include <common_properties.h>
+#include <fc_layer_cl.h>
+#include <layer_context.h>
+#include <lazy_tensor.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <node_exporter.h>
+#include <util_func.h>
+
+std::string fc_sgemv_cl_kernel_ =
```

Comment:
can we make a binary cl kernel to reduce online compilation time?

Reply:
This implementation is present in the current code. So, `program.CreateCLProgram` call from `RunLayerContext::clCreateKernel` will create kernel binaries for the first run and re-use them from the next runs.

Reply:
We might need to set the proper directory to save the CL Kernel Binary then.

Reply:
Right. As of now it is being set by `opencl-kernel-path` of `meson_options`.